src/MultiplayerMod/Multiplayer/CoreOperations/PlayersManagement/PlayersManagementController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using MultiplayerMod.Core.Events;
+using MultiplayerMod.Core.Logging;
+using MultiplayerMod.Multiplayer.CoreOperations.Events;
+using MultiplayerMod.Multiplayer.CoreOperations.PlayersManagement.Commands;
+using MultiplayerMod.Multiplayer.Players;
+using MultiplayerMod.Multiplayer.World;
+using MultiplayerMod.Network;
+
+namespace MultiplayerMod.Multiplayer.CoreOperations.PlayersManagement;
+
+[UsedImplicitly]
+public class PlayersManagementController {
+
+    private readonly Core.Logging.Logger log = LoggerFactory.GetLogger<PlayersManagementController>();
+
+    private readonly IMultiplayerServer server;
+    private readonly IMultiplayerClient client;
+
+    private readonly WorldManager worldManager;
+    private readonly MultiplayerGame multiplayer;
+
+    private readonly IPlayerProfileProvider profileProvider;
+
+    private readonly Dictionary<IMultiplayerClientId, PlayerIdentity> identities = new();
+
+    public PlayersManagementController(
+        IMultiplayerServer server,
+        IMultiplayerClient client,
+        EventDispatcher events,
+        IPlayerProfileProvider profileProvider,
+        WorldManager worldManager,
+        MultiplayerGame multiplayer
+    ) {
+        this.client = client;
+        this.profileProvider = profileProvider;
+        this.server = server;
+        this.worldManager = worldManager;
+        this.multiplayer = multiplayer;
+
+        server.ClientDisconnected += OnClientDisconnected;
+        events.Subscribe<ClientInitializationRequestEvent>(OnClientInitializationRequested);
```

Comment:
Also feels slightly wrong about those 2 events being here.

And what is a purpose of identites map? Right now I don't see any read access to it (beside assert).